### PR TITLE
Adding nodemanager ingestion project tagging tests

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Adding API level integration tests for node manager ingestion project tagging.

### :chains: Related Resources

#2193

### :+1: Definition of Done

Have a test for each possible attribute for compliance project update tagging.

### :athletic_shoe: How to Build and Test the Change
Check the buildkite tests

### :white_check_mark: Checklist

- [x] Tests added/updated?
- [ ] Docs added/updated?

### :camera: Screenshots, if applicable

### CODE REVIEW

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
+import { describeIfIAMV2p1 } from '../../constants';
+import { uuidv4 } from '../../helpers';
+
+describeIfIAMV2p1('Nodemanager ingestion project tagging', () => {
+  const cypressPrefix = 'test-nodemanager-ingestion';
+
+
+  const projectsWithRule = [
+    // This test is commented out because there is a current limit of 6 projects allowed
```
Comment (blakestier) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]: We can uncomment now 🎉 https://github.com/chef/automate/pull/2308

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+  it('tagging 7 projects with all the attributes on a compliance and client runs node', () => {
+    const complianceNodeId = uuidv4();
+    const clientRunsNodeId = uuidv4();
+    const reportId = uuidv4();
+    const nodeName = `${cypressPrefix}-${Cypress.moment().format('MMDDYYhhmm')}`;
+    const start = Cypress.moment().utc().subtract(3, 'day').startOf('day').format();
+    const end = Cypress.moment().utc().endOf('day').format();
+
+    // Ingest a InSpec report with attribues that match all the projects
```
Comment (blakestier) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]:
```suggestion
    // Ingest a InSpec report with attributes that match all the projects
```

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+      report.report_uuid = reportId;
+      cy.request({
+        headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+        method: 'POST',
+        url: '/data-collector/v0',
+        body: report
+      });
+    });
+
+    // Ingest a node with attribues that match all the projects
```
Comment (blakestier) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]:
```suggestion
    // Ingest a node with attributes that match all the projects
```

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+        },
+        method: 'POST',
+        url: '/api/v0/nodes/search',
+        body: {
+          order: 'DESC',
+          page: 1,
+          per_page: 100
+        }
+      }).then((response) => {
+        expect(response.body.nodes.length).to.greaterThan(0);
```
Comment (blakestier) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]: does this `to` need a `be`?

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+    if (node.id === nodeId) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function waitForNodemanagerNode(nodeId: string, maxRetries: number) {
+  cy.request({
```
Comment (blakestier) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]: starting to think we should have a helper that makes the request with the token 🤔

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts (hunk 4 above)
Comment (lancewf) [reply e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]: That works also 🤷‍♂

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts (hunk 5 above)
Comment (lancewf) [reply e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.ts]: That would be helpful 😄

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.spec.ts [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+
+    // wait for the report to be ingested
+    waitForNodemanagerNode(complianceNodeId, 30);
+    // wait for the client run report to be ingested
+    waitForNodemanagerNode(clientRunsNodeId, 30);
+  });
+
+  after(() => cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']));
+
+  // Ensure the both the compliance and client run nodes' tags are updated
```
Comment (msorens) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.spec.ts]:
```suggestion
  // Ensure that both the compliance and client run nodes' tags are updated
```

File: e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.spec.ts [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+});
+
+function nodeExist(nodeId: string, nodes: any[]): boolean {
+  for (const node of nodes ) {
+    if (node.id === nodeId) {
+      return true;
+    }
+  }
+
+  return false;
```
Comment (msorens) [inline e2e/cypress/integration/api/iam/09_projects_api_nodemanager_ingestion.spec.ts]: Can be written more concisely:
```
return nodes.some(node => node.id === nodeId);
```

**Pull request merged**